File: activityloghelper/src/main/java/com/example/administrator/activityloghelper/LogHelperApplication.java
package com.example.administrator.activityloghelper;

import android.app.Application;
import android.os.Build;
import android.util.Log;

import java.lang.reflect.Method;

/**
 * Created by wangyi on 2017/2/8.
 */

public class LogHelperApplication extends Application {

    private static LogHelperApplication sInstance;

    public LogHelperApplication() {
        sInstance = this;
    }

    public static LogHelperApplication getInstance() {
        return sInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
    }


    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        Log.d("wangyi", "onTrimMemory: " + level);
    }


}

File: activityloghelper/src/main/java/com/example/administrator/activityloghelper/LogHelperApplication.kt
package com.example.administrator.activityloghelper

import android.app.Application
import android.content.Context
import android.util.Log

/**
 * Created by wangyi on 2017/2/8.
 *
 */
class LogHelperApplication : Application() {

    //在默认构造函数进行赋值
    init {
        instance = this
    }

    companion object {

        private lateinit var instance: LogHelperApplication

        @JvmStatic
        fun getInstance(): Context {
            return instance
        }
    }

    override fun onCreate() {
        super.onCreate()
    }


    //kotlin 不需要返回void
    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)
        Log.d("wangyi", "onTrimMemory: " + level)
    }


}
